src/lib/linux-api/src/signal.rs

```diff
@@ -0,0 +1,1197 @@
+    pub const SIGXCPU: Self = Self(bindings::LINUX_SIGXCPU as i32);
+    pub const SIGXFSZ: Self = Self(bindings::LINUX_SIGXFSZ as i32);
+    pub const SIGVTALRM: Self = Self(bindings::LINUX_SIGVTALRM as i32);
+    pub const SIGPROF: Self = Self(bindings::LINUX_SIGPROF as i32);
+    pub const SIGWINCH: Self = Self(bindings::LINUX_SIGWINCH as i32);
+    pub const SIGIO: Self = Self(bindings::LINUX_SIGIO as i32);
+    pub const SIGPWR: Self = Self(bindings::LINUX_SIGPWR as i32);
+    pub const SIGSYS: Self = Self(bindings::LINUX_SIGSYS as i32);
+
+    pub const STANDARD_MAX: Self = Self::SIGSYS;
+
+    pub const SIGRT_MIN: Self = Self(bindings::LINUX_SIGRTMIN as i32);
+    // According to signal(7). bindgen fails to bind this one.
+    pub const SIGRT_MAX: Self = Self(64);
+
+    pub const MIN: Self = Self(1);
+    pub const MAX: Self = Self::SIGRT_MAX;
+
+    const fn const_alias(from: u32, to: Self) -> Self {
+        if from as i32 != to.0 {
+            // Can't use a format string here since this function is `const`
+            panic!("Incorrect alias")
+        }
+        to
+    }
+    pub const SIGIOT: Self = Self::const_alias(bindings::LINUX_SIGIOT, Self::SIGABRT);
+    pub const SIGPOLL: Self = Self::const_alias(bindings::LINUX_SIGPOLL, Self::SIGIO);
+    pub const SIGUNUSED: Self = Self::const_alias(bindings::LINUX_SIGUNUSED, Self::SIGSYS);
+
+    pub fn is_realtime(&self) -> bool {
+        (i32::from(Self::SIGRT_MIN)..=i32::from(Self::SIGRT_MAX)).contains(&self.0)
+    }
+
+    pub const fn as_i32(&self) -> i32 {
+        self.0
+    }
+}
+
+bitflags::bitflags! {
+    #[repr(transparent)]
+    #[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
+    pub struct SigActionFlags: u64 {
+        const NOCLDSTOP = bindings::LINUX_SA_NOCLDSTOP as u64;
+        const NOCLDWAIT = bindings::LINUX_SA_NOCLDWAIT as u64;
+        const NODEFER = bindings::LINUX_SA_NODEFER as u64;
+        const ONSTACK = bindings::LINUX_SA_ONSTACK as u64;
+        const RESETHAND = bindings::LINUX_SA_RESETHAND as u64;
+        const RESTART = bindings::LINUX_SA_RESTART as u64;
+        const RESTORER = bindings::LINUX_SA_RESTORER as u64;
+        const SIGINFO = bindings::LINUX_SA_SIGINFO as u64;
+    }
+}
+// We can't derive this since the bitflags field uses an internal type.
+unsafe impl VirtualAddressSpaceIndependent for SigActionFlags {}
+
+/// Describes how a signal was sent.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum SigInfoCode {
+    Si(SigInfoCodeSi),
+    Ill(SigInfoCodeIll),
+    Fpe(SigInfoCodeFpe),
+    Segv(SigInfoCodeSegv),
+    Bus(SigInfoCodeBus),
+    Trap(SigInfoCodeTrap),
+    Cld(SigInfoCodeCld),
+    Poll(SigInfoCodePoll),
+    Sys(SigInfoCodeSys),
+}
+
+#[derive(Debug, Copy, Clone)]
+pub struct SigInfoCodeFromRawError {
+    pub code: i32,
+    pub signo: i32,
+}
+
+impl SigInfoCode {
+    fn try_from_raw(raw_code: i32, raw_signo: i32) -> Result<Self, SigInfoCodeFromRawError> {
```

Comment:
I think this could use a comment explaining what the function is used for.

Reply:
Done